contracts/cw721-metadata-onchain/src/lib.rs

```diff
@@ -4,11 +4,29 @@ use serde::{Deserialize, Serialize};
 use cosmwasm_std::Empty;
 pub use cw721_base::{ContractError, InstantiateMsg, MintMsg, MinterResponse, QueryMsg};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-pub struct Extension {
-    pub metadata_uri: String,
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Trait {
+    pub display_type: Option<String>,
+    pub trait_type: String,
+    pub value: String,
 }
 
+// see: https://docs.opensea.io/docs/metadata-standards
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
```

Comment:
As an example, this makes sense, that we store it in the exact same format that OpenSea has it.

However, if we build more complex NFTs, we would want to keep this `Metadata` struct as the query result from NFT Info (or have a custom query that just returns it). And maybe store it in a different format, that is more tied to the business case and just transform it to this format for display.

I am thinking Royalties. It is possible to store them in a Trait, but sub-optimal.
However, they could easily be converted to a Trait when serving a query

Reply:
I suppose there's also the question, since these are all Options anyway, whether we include the royalty fields for 2981 by default, and then simply leave it up to the implementer to define the custom queries that make up the rest of that royalty interface... 🤔